src/DGmethods/DGmodel.jl

```diff
@@ -242,6 +242,47 @@ function nodal_update_aux!(f!, dg::DGModel, m::BalanceLaw, Q::MPIStateArray,
           knl_nodal_update_aux!(m, Val(dim), Val(polyorder), f!,
                           Q.data, dg.auxstate.data, dg.diffstate.data, t,
                           topology.realelems))
+
+end
+
+"""
+    cfl(wavespeed_dt, dg::DGModel, m::BalanceLaw, Q::MPIStateArray,
+        direction=EveryDirection())
+
+Returns an approximation of the CFL number `(c * Δt)/Δx` where `c` is the wave
+speed of interest.
+
+The function `wavespeed_dt` should compute `c * Δt` given the `state`, `aux`
+and `diffusive` variables.  The `direction` controls which reference
+directions are considered when computing the minimum node distance `Δx`.
+"""
+function cfl(wavespeed_dt, dg::DGModel, m::BalanceLaw, Q::MPIStateArray,
+             direction=EveryDirection())
+  grid = dg.grid
+  topology = grid.topology
+  nrealelem = length(topology.realelems)
+
+  if nrealelem > 0
+    N = polynomialorder(dg.grid)
+    dim = dimensionality(grid)
+    Nq = N + 1
+    Nqk = dim == 2 ? 1 : Nq
+    device = grid.vgeo isa Array ? CPU() : CUDA()
+    local_cfl = similar(grid.vgeo, Nq^dim, nrealelem)
+    @launch(device, threads=(Nq, Nq, Nqk), blocks=nrealelem,
+            Grids.knl_min_neighbor_distance!(Val(N), Val(dim), direction,
+                                             local_cfl, grid.vgeo,
+                                             topology.realelems))
+    @launch(device, threads=(Nq*Nq*Nqk,), blocks=nrealelem,
+            knl_local_cfl!(m, Val(dim), Val(N), local_cfl, wavespeed_dt,
+                           Q.data, dg.auxstate.data, dg.diffstate.data,
+                           topology.realelems))
+    locmax = maximum(local_cfl)
+  else
+    locmax = typemax(eltype(Q))
```

Comment:
when does this branch get used?

Reply:
When an MPI rank doesn't have any elements.

Reply:
then should it be `typemin` (since otherwise this will become the result from `Allreduce`)?